esphome/components/zyaura/sensor.py

```diff
@@ -0,0 +1,41 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import pins
+from esphome.components import sensor
+from esphome.const import CONF_ID, CONF_CLOCK_PIN, CONF_DATA_PIN, CONF_UPDATE_INTERVAL, \
+    CONF_CO2, CONF_TEMPERATURE, CONF_HUMIDITY, \
+    UNIT_PARTS_PER_MILLION, UNIT_CELSIUS, UNIT_PERCENT, \
+    ICON_PERIODIC_TABLE_CO2, ICON_THERMOMETER, ICON_WATER_PERCENT
+from esphome.cpp_helpers import gpio_pin_expression
+
+zyaura_ns = cg.esphome_ns.namespace('zyaura')
+ZyAuraSensor = zyaura_ns.class_('ZyAuraSensor', cg.PollingComponent)
+
+CONFIG_SCHEMA = cv.Schema({
+    cv.GenerateID(): cv.declare_id(ZyAuraSensor),
+    cv.Required(CONF_CLOCK_PIN): pins.gpio_input_pin_schema,
```

Comment:
```suggestion
    cv.Required(CONF_CLOCK_PIN): cv.All(pins.internal_gpio_input_pin_schema, pins.validate_has_interrupt),
```

We should validate that the pin has interrupt capabilities here because we're using interrupts on it.